src/main/java/ca/teamdman/sfm/common/capabilityprovidermapper/ae2/InterfaceCapabilityProviderMapper.java

```diff
@@ -0,0 +1,446 @@
+        @Nullable
+        InterfaceBlockEntity getInterface() {
+            return interfaceAt(this.level, this.pos);
+        }
+
+        @Nullable
+        IEnergyService getEnergy() {
+            var in = this.getInterface();
+            if (in == null) {
+                return null;
+            }
+
+            var grid = in.getMainNode().getGrid();
+            if (grid == null) {
+                return null;
+            }
+
+            return grid.getEnergyService();
+        }
+
+        @NotNull
+        LazyOptional<IStorageMonitorableAccessor> getCapability() {
+            var in = this.getInterface();
+            if (in == null) {
+                return LazyOptional.empty();
+            }
+
+            if (!in.getConfig().isEmpty() || in.getMainNode() == null || in.getGridNode() == null || !in.getGridNode().isActive()) {
+                return LazyOptional.empty();
+            }
+
+            return in.getCapability(Capabilities.STORAGE_MONITORABLE_ACCESSOR);
+        }
+
+        <T> T withStorage(Function<MEStorage, T> callback) {
+            var cap = this.getCapability();
+            if (cap.isPresent()) {
+                //noinspection DataFlowIssue
+                return callback.apply(cap.map(c -> c.getInventory(IActionSource.empty())).orElse(null));
+            }
+            return null;
+        }
+
+        <T> T withBaseItemHandler(Function<IItemHandler, T> callback) {
+            var in = this.getInterface();
+            if (in == null) {
+                return null;
+            }
+
+            var maybeCap = in.getCapability(ForgeCapabilities.ITEM_HANDLER);
+            if (maybeCap.isPresent()) {
+                //noinspection DataFlowIssue
+                return callback.apply(maybeCap.orElse(null));
+            }
+            return null;
+        }
+
+        <T> T withBaseFluidHandler(Function<IFluidHandler, T> callback) {
+            var in = this.getInterface();
+            if (in == null) {
+                return null;
+            }
+
+            var maybeCap = in.getCapability(ForgeCapabilities.FLUID_HANDLER);
+            if (maybeCap.isPresent()) {
+                //noinspection DataFlowIssue
+                return callback.apply(maybeCap.orElse(null));
+            }
+            return null;
+        }
+
+        @Override
+        public int getSlots() {
+            Integer slots = this.withStorage(s -> {
+                int i = 0;
+                for (var stored : s.getAvailableStacks()) {
+                    if (stored.getKey() instanceof AEItemKey) {
+                        i++;
+                    }
+                }
+                return i;
+            });
+
+            if (slots == null) {
+                slots = this.withBaseItemHandler(IItemHandler::getSlots);
+            }
+
+            return slots == null ? 0 : slots;
+        }
+
+        @Override
+        public @NotNull ItemStack getStackInSlot(int slot) {
```

Comment:
```suggestion
        public ItemStack getStackInSlot(int slot) {
```
the package file should be adding notnullbydefault to everything